src/services/rollouts.js

```diff
@@ -832,4 +834,176 @@ module.exports = class RolloutsHelper {
 			throw error
 		}
 	}
+
+	/**
+	 * Update program rollout
+	 * @method
+	 * @name updateProgramRollout
+	 * @param {Integer} programId - program Id
+	 * @param {Object} programData - Program data object
+	 * @param {String} userId - The ID of the user
+	 * @param {String} orgId - The ID of the Organization
+	 * @returns {Integer} - program rollout id
+	 */
+	static async updateProgramRollout(programId, programData, userId, orgId) {
+		try {
+			// fetch the resource ids from the program
+			const programResourceIds = programData.resources.map((resource) => resource.id)
+
+			// fetch rollout data of program and resources
+			const fetchRollouts = await rolloutQueries.findAll(
+				{
+					resource_id: {
+						[Op.in]: [programId, ...programResourceIds],
+					},
+					user_id: userId,
+					organization_id: orgId,
+				},
+				{
+					attributes: ['id', 'resource_type', 'resource_id'],
+				}
+			)
+			let resourceRolloutResourceIdMap = {} // initialise rollout id resource mapping
+			let programRolloutId // initialise variable for program rollout id
+
+			// seggregate program rollout id and create rollout id resource mapping
+			fetchRollouts.forEach((rollout) => {
+				if (rollout.type != common.RESOURCE_TYPE_PROGRAM) {
+					resourceRolloutResourceIdMap[rollout.resource_id] = rollout.id
+				} else if (rollout.type == common.RESOURCE_TYPE_PROGRAM && rollout.resource_id == programId) {
+					programRolloutId = rollout.id
+				}
+			})
+
+			// update rollout variable
+			let rolloutUpdate = {
+				start_date: programData?.meta?.start_date || '',
+				end_date: programData?.meta?.end_date || '',
+				targeting_criteria: programData?.targeting_criteria,
+				updated_at: new Date(),
+			}
+			// prepare resources for program rollout update
+			programData.resources.forEach((resource) => {
+				rolloutUpdate.resources.push(resource)
+			})
+			// create a promise variable and add program rollout update
+			let rolloutUpdatePromise = [this.update(programRolloutId, rolloutUpdate, userId, orgId)]
```

Comment:
@adithyadinesh0412  if program and solution rollout is running parallely when to update parent id